src/fides/api/models/privacy_preference.py

```diff
@@ -686,11 +711,23 @@ class CurrentPrivacyPreference(LastSavedMixin, Base):
             "privacy_notice_id",
             name="fides_user_device_identity_privacy_notice",
         ),
-        UniqueConstraint("provided_identity_id", "purpose", name="identity_purpose"),
+        UniqueConstraint(
+            "provided_identity_id", "purpose_consent", name="identity_purpose_consent"
+        ),
```

Comment:
Removed old unique constraints and added new ones.  These help enforce that only a single preference for a given identity and preference type are stored in these tables.

Please help double check that I don't have copy-paste errors here!

There are two identity types against which consent can be saved, and 6 new columns added.